ir_x86.dasc

```diff
@@ -11541,9 +11549,17 @@ next_block:;
 		return NULL;
 	}
 
+	if (data.dasm_state->status != DASM_S_OK) {
+		IR_ASSERT(data.dasm_state->status == DASM_S_NOMEM);
+		dasm_free(&data.dasm_state);
+		ctx->data = NULL;
+		ctx->status = IR_ERROR_TOO_LARGE;
+		return NULL;
+	}
+
 	ret = dasm_link(&data.dasm_state, size_ptr);
 	if (ret != DASM_S_OK) {
-		IR_ASSERT(0);
+		IR_ASSERT(ret == DASM_S_NOMEM);
 		dasm_free(&data.dasm_state);
 		ctx->data = NULL;
 		ctx->status = IR_ERROR_LINK;
```

Comment:
IR_ASSERT/ctx->status behavior is still inconsistent.
At this point we may return IR_ERROR_LINK instead of IR_ERROR_TOO_LARGE, in other places we may return incorrect IR_ERROR_TOO_LARGE in case ASSERTS are ignored.

Reply:
I'm not sure what's the best way to handle this.

I assumed that `data.dasm_state->status` can only be `DASM_S_OK` or `DASM_S_NOMEM`, as other codes are only emitted in case of a bug, and some are only emitted when `IR_DEBUG` / `DASM_CHECKS` are defined.

If that's not the case, I see two possibilities:
 * Check all `DASM_S_` codes here and update ctx->status accordingly
 * Set ctx->status to a more generic code such as `IR_ERROR_CODE_GEN` instead of `IR_ERROR_TOO_LARGE`

Reply:
> I assumed that data.dasm_state->status can only be DASM_S_OK or DASM_S_NOMEM, as other codes are only emitted in case of a bug

Right. The original ``ASSERT(0)`` was useful to catch bugs in DEBUG mode.

Ignoring DASM_S_NOMEM makes sense, but ctx->status should be set to the valid error code
```c
ctx->status  = (ret ==  DASM_S_NOMEM) ? IR_ERROR_TOO_LARGE : IR_ERROR_LINK;
```